PortSpeedTestTemplate/NetworkCardSpeedTestTemplate.cs

```diff
@@ -0,0 +1,146 @@
+﻿//Company's property 2024
+
+namespace PortSpeedTestTemplate
+{
+    /// <summary>
+    /// Network card speed test template
+    /// </summary>
+    public abstract class NetworkCardSpeedTestTemplate : Program
+    {
+        #region Properties
+
+        /// <summary>
+        /// Property
+        /// </summary>
+        protected virtual List<(byte mode, List<(int port, int speed)>)> TestData { get; set; }
+
+        #endregion
+
+        #region Public methods
+
+        /// <summary>
+        /// Prepare
+        /// </summary>
+        /// <returns></returns>
+        public abstract bool Prepare();
+
+        /// <summary>
+        /// Runner
+        /// </summary>
+        public abstract void Run();
+
+        /// <summary>
+        /// Cleaner after tests
+        /// </summary>
+        /// <returns></returns>
+        public abstract bool Clean();
+
+        #endregion
+
+        #region Protected methods
+
+        /// <summary>
+        /// Active card mode
+        /// </summary>
+        /// <param name="mode">
+        /// Card mode
+        /// </param>
+        protected static void CheckActiveCardMode(byte mode)
```

Comment:
I miss it, so leave it like it is. But for the future your methods will be not static and should be bool